python/src/aica_api/client.py

```diff
@@ -81,18 +91,44 @@ def wrapper(self, *args, **kwargs):
 
     def api_version(self) -> Union[str, None]:
         """
-        Get the version of the AICA API server
+        Get the specific version the AICA API server as a sub-package of AICA Core
+
+        :return: The version of the API server or None in case of connection failure
+        """
+        try:
+            api_server_version = self.license().json()['signed_packages']['aica_api_server']
+            self._logger.debug(f'AICA API server version identified as {api_server_version}')
+            return api_server_version
+        except requests.exceptions.RequestException:
+            self._logger.error(f'Error connecting to the API server at {self._address}! '
+                               f'Check that AICA Core is running and configured with the right address.')
+        except requests.exceptions.JSONDecodeError:
+            self._logger.error(f'Error getting version details! Expected JSON data in the response body.')
+        except KeyError as e:
+            self._logger.error(
+                f'Error getting version details! Expected a map of `signed_packages` to include `aica_api_server`: {e}')
+        return None
+
+    def core_version(self) -> Union[str, None]:
+        """
+        Get the version of the AICA Core
 
-        :return: The version of the API server or None in case of
+        :return: The version of the AICA core or None in case of connection failure
         """
+        core_version = None
         try:
-            self._api_version = requests.get(f'{self._address}/version').json()
-            self._logger.debug(f'API server version identified as {self._api_version}')
+            core_version = requests.get(f'{self._address}/version').json()
         except requests.exceptions.RequestException:
             self._logger.error(f'Error connecting to the API server at {self._address}! '
-                               f'Check that the AICA container is running and configured with the right address.')
-            self._api_version = None
-        return self._api_version
+                               f'Check that AICA Core is running and configured with the right address.')
+
+        if not semver.Version.is_valid(core_version):
```

Comment:
This still doesn't work because the `is_valid` requires a string, and in case of a custom core, the core_version is a dict

Reply:
ugh the problem is that I lost my previous changes and couldn't remember exactly how I did it the first time. My fault for not testing. I did actually do try / except TypeError around `parse_version_info` the first time but then thought that Version.isValid is probably nicer, forgetting that the type error will still happen